app/views/business_categories/_as_list.html.haml

```diff
@@ -1,4 +1,4 @@
 .post-category
   %ul.post-categories
-    - @company.business_categories.distinct.each do | business_category |
+    - categories.distinct.each do | business_category |
```

Comment:
Why not use the method you defined in Company to simplify this?  

In this file:
```
.post-category
  %ul.post-categories
    - categories_names.each do | name |
      %li= name
```

In `views/companies/edit.html.haml`:
```
= render 'business_categories/as_list',
             categories_names: @company.categories_names
```
Similarly in the companies `show` view.